osidb/models/affect.py

```diff
@@ -172,6 +172,16 @@ class Meta:
     def __str__(self):
         return str(self.uuid)
 
+    def save(self, *args, **kwargs):
+        if self.purl and not self.ps_component:
+            try:
+                ps_component = PackageURL.from_string(self.purl).name
+            except Exception as exc:
+                raise ValidationError(f"Invalid purl '{self.purl}': {exc}")
```

Comment:
There should be some human-readable affect identification so when a user creates/updates multiple affects and then sees an error it is clear which one was invalid.